### ISSUE TITLE
mingw build failure

### ISSUE THREAD

**Issue by rorgoroth** [dhewm/dhewm3#505]
Hi, 
I've gotten this error building with mingw64, I let ninja run past fails and this is all that's left before a successful build.

````llvm
[1/6] Building CXX object CMakeFiles/dhewm3ded.dir/tools/debugger/debugger.cpp.obj
[2/6] Building CXX object CMakeFiles/dhewm3ded.dir/tools/debugger/DebuggerServer.cpp.obj
[3/6] Building CXX object CMakeFiles/dhewm3.dir/tools/debugger/debugger.cpp.obj
[4/6] Building CXX object CMakeFiles/dhewm3.dir/tools/debugger/DebuggerServer.cpp.obj
ninja: job failed: /mingw64/bin/x86_64-w64-mingw32-g++ -DD3_ARCH=\"x86_64\" -DD3_OSTYPE=\"windows\" -DD3_SIZEOFPTR=8 -DID_DEDICATED -DWINVER=0x0501 -D_WIN32_WINNT=0x0501 -D__DOOM_DLL__ -I/mingw64/mingw/include/AL -I/mingw64/mingw/include/SDL2 -I/home/ray/dhewm3/build -I/home/ray/dhewm3/neo -DNDEBUG -O2 -fno-math-errno -fno-trapping-math -fomit-frame-pointer -pipe -Wall -fdiagnostics-color=always -fno-strict-aliasing -ffp-contract=off -fvisibility=hidden -Wno-sign-compare -Wno-switch -Woverloaded-virtual -Wno-class-memaccess -MD -MT CMakeFiles/dhewm3ded.dir/tools/debugger/DebuggerServer.cpp.obj -MF CMakeFiles/dhewm3ded.dir/tools/debugger/DebuggerServer.cpp.obj.d -o CMakeFiles/dhewm3ded.dir/tools/debugger/DebuggerServer.cpp.obj -c /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.cpp
In file included from /mingw64/mingw/include/SDL2/SDL_cpuinfo.h:70,
                 from /mingw64/mingw/include/SDL2/SDL.h:38,
                 from /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.h:35,
                 from /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.cpp:41:
/home/ray/dhewm3/neo/idlib/Str.h:44:33: error: declaration of 'static int idStr::Cmp(const char*, const char*)' outside of class is not definition [-fpermissive]
   44 | #define strcmp                  idStr::Cmp              // use_idStr_Cmp
      |                                 ^~~~~
ninja: job failed: /mingw64/bin/x86_64-w64-mingw32-g++ -DD3_ARCH=\"x86_64\" -DD3_OSTYPE=\"windows\" -DD3_SIZEOFPTR=8 -DWINVER=0x0501 -D_WIN32_WINNT=0x0501 -D__DOOM_DLL__ -I/mingw64/mingw/include/AL -I/mingw64/mingw/include/SDL2 -I/home/ray/dhewm3/build -I/home/ray/dhewm3/neo -DNDEBUG -O2 -fno-math-errno -fno-trapping-math -fomit-frame-pointer -pipe -Wall -fdiagnostics-color=always -fno-strict-aliasing -ffp-contract=off -fvisibility=hidden -Wno-sign-compare -Wno-switch -Woverloaded-virtual -Wno-class-memaccess -MD -MT CMakeFiles/dhewm3.dir/tools/debugger/debugger.cpp.obj -MF CMakeFiles/dhewm3.dir/tools/debugger/debugger.cpp.obj.d -o CMakeFiles/dhewm3.dir/tools/debugger/debugger.cpp.obj -c /home/ray/dhewm3/neo/tools/debugger/debugger.cpp
In file included from /mingw64/mingw/include/SDL2/SDL_cpuinfo.h:70,
                 from /mingw64/mingw/include/SDL2/SDL.h:38,
                 from /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.h:35,
                 from /home/ray/dhewm3/neo/tools/debugger/debugger.cpp:37:
/home/ray/dhewm3/neo/idlib/Str.h:44:33: error: declaration of 'static int idStr::Cmp(const char*, const char*)' outside of class is not definition [-fpermissive]
   44 | #define strcmp                  idStr::Cmp              // use_idStr_Cmp
      |                                 ^~~~~
ninja: job failed: /mingw64/bin/x86_64-w64-mingw32-g++ -DD3_ARCH=\"x86_64\" -DD3_OSTYPE=\"windows\" -DD3_SIZEOFPTR=8 -DID_DEDICATED -DWINVER=0x0501 -D_WIN32_WINNT=0x0501 -D__DOOM_DLL__ -I/mingw64/mingw/include/AL -I/mingw64/mingw/include/SDL2 -I/home/ray/dhewm3/build -I/home/ray/dhewm3/neo -DNDEBUG -O2 -fno-math-errno -fno-trapping-math -fomit-frame-pointer -pipe -Wall -fdiagnostics-color=always -fno-strict-aliasing -ffp-contract=off -fvisibility=hidden -Wno-sign-compare -Wno-switch -Woverloaded-virtual -Wno-class-memaccess -MD -MT CMakeFiles/dhewm3ded.dir/tools/debugger/debugger.cpp.obj -MF CMakeFiles/dhewm3ded.dir/tools/debugger/debugger.cpp.obj.d -o CMakeFiles/dhewm3ded.dir/tools/debugger/debugger.cpp.obj -c /home/ray/dhewm3/neo/tools/debugger/debugger.cpp
In file included from /mingw64/mingw/include/SDL2/SDL_cpuinfo.h:70,
                 from /mingw64/mingw/include/SDL2/SDL.h:38,
                 from /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.h:35,
                 from /home/ray/dhewm3/neo/tools/debugger/debugger.cpp:37:
/home/ray/dhewm3/neo/idlib/Str.h:44:33: error: declaration of 'static int idStr::Cmp(const char*, const char*)' outside of class is not definition [-fpermissive]
   44 | #define strcmp                  idStr::Cmp              // use_idStr_Cmp
      |                                 ^~~~~
ninja: job failed: /mingw64/bin/x86_64-w64-mingw32-g++ -DD3_ARCH=\"x86_64\" -DD3_OSTYPE=\"windows\" -DD3_SIZEOFPTR=8 -DWINVER=0x0501 -D_WIN32_WINNT=0x0501 -D__DOOM_DLL__ -I/mingw64/mingw/include/AL -I/mingw64/mingw/include/SDL2 -I/home/ray/dhewm3/build -I/home/ray/dhewm3/neo -DNDEBUG -O2 -fno-math-errno -fno-trapping-math -fomit-frame-pointer -pipe -Wall -fdiagnostics-color=always -fno-strict-aliasing -ffp-contract=off -fvisibility=hidden -Wno-sign-compare -Wno-switch -Woverloaded-virtual -Wno-class-memaccess -MD -MT CMakeFiles/dhewm3.dir/tools/debugger/DebuggerServer.cpp.obj -MF CMakeFiles/dhewm3.dir/tools/debugger/DebuggerServer.cpp.obj.d -o CMakeFiles/dhewm3.dir/tools/debugger/DebuggerServer.cpp.obj -c /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.cpp
In file included from /mingw64/mingw/include/SDL2/SDL_cpuinfo.h:70,
                 from /mingw64/mingw/include/SDL2/SDL.h:38,
                 from /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.h:35,
                 from /home/ray/dhewm3/neo/tools/debugger/DebuggerServer.cpp:41:
/home/ray/dhewm3/neo/idlib/Str.h:44:33: error: declaration of 'static int idStr::Cmp(const char*, const char*)' outside of class is not definition [-fpermissive]
   44 | #define strcmp                  idStr::Cmp              // use_idStr_Cmp
      |                                 ^~~~~
ninja: subcommands failed
````

Any hints/help/fixes appreciated!

**Comment by DanielGibson** [comment]
urgh

I'll fix that later, as a workaround just comment out the `#define strcmp idStr::Cmp` line

**Comment by DanielGibson** [comment]
should be fixed with 07985cc5ea1bc, can you try if it works for you as well?
